src/hip_module.cpp

```diff
@@ -271,6 +276,13 @@ hipError_t ihipModuleLaunchKernel(hipFunction_t f, uint32_t globalWorkSizeX,
         }
 
         ihipPostLaunchKernel(f->_name.c_str(), hStream, lp);
+
+#if (__hcc_workweek__ >= 19213)
+        if (lockHSAQueue) {
+            lp.av->release_locked_hsa_queue();
```

Comment:
I think this should happen after all the kernels have been enqueued. Otherwise there is no guarantee of atomicity.

Reply:
I made changes to call release_locked_hsa_queue for each respective stream after all the kernels have been en queued.